lib/main.js

```diff
@@ -64,39 +64,56 @@ function weave(format) {
     let outpath = edpath.split('.')
     outpath.pop()
     outpath = outpath.join('.') + '.' + format
+    let doctype = 'md2' + format
 
     let weaveCommand  = 'using Weave;'
     weaveCommand += 'try;'
-    weaveCommand += `weave("${edpath}", doctype="md2${format}"); 1;`
+    weaveCommand += `weave("${edpath}", doctype="${doctype}"); 1;`
     weaveCommand += 'catch err;'
     weaveCommand += '@error("Weaving failed.", error=err); 0;'
     weaveCommand += 'end'
 
+    let outPathCommand = 'using Weave: separate_header_text, parse_header, specific_options!;'
+    outPathCommand += `headertext, = separate_header_text(read("${edpath}", String));`
+    outPathCommand += 'header = parse_header(headertext);'
+    outPathCommand += 'weave_options = get(header, "weave_options", Dict());'
+    outPathCommand += `format_options = specific_options!(weave_options, "${doctype}");`
+    outPathCommand += 'actual_options = isnothing(format_options) ? weave_options : format_options;'
+    outPathCommand += 'get(actual_options, "out_path", "")'
+
```

Comment:
I guess the easiest way will be to use the return value of `weave`; how about changing `weaveCommand` so that it will do:
```julia
return try
  weave("${edpath}", doctype = "${doctype}")
catch err
  @error "Weaving faild." error = err
  0
end
```

Reply:
Aha, I didn't even notice that that was the output of the command :+1: , cheers